src/io/procfs.rs

```diff
@@ -207,6 +207,17 @@ fn is_mountpoint(file: BorrowedFd<'_>) -> bool {
     }
 }
 
+// Open a directory in `/proc`, mapping all errors to `-ENOTSUP`
```

Comment:
While the value of the ultimate error is indeed `-ENOTSUP` with the negation, that's a lower level of abstraction from the perspective of this code. This code should think of it as `io::Error::NOTSUP` or similar.

Also, this comment can be a triple-slash comment, since it is still a documentation comment, even though it's a private function.